src/pages/Join/Join.js

```diff
@@ -1,8 +1,150 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
 import './Join.scss';
 
 const Join = () => {
-  return <div>Join</div>;
+  const [joinValue, setJoinValue] = useState({
+    id: '',
+    pw: '',
+    email: '',
+  });
+
+  const navigate = useNavigate();
+
+  const onJoin = () => {
+    const { id, pw, email } = joinValue;
+    const validId = id !== '' && 2 < id.length < 16;
+    const validPw = pw !== '' && 7 < pw.length < 16;
+    const validEmail = email !== '' && pw.includes('@', '.');
+
+    if (!validId) {
+      alert(
+        '[ 아이디 ] 는(은) 공백없는 3~15자의 소문자와 숫자를 조합하여 입력해주세요.'
+      );
+    } else if (!validPw) {
+      alert(
+        '[ 비밀번호 ] 는(은) 공백없는 8~15자의 영문/특수문자/숫자를 조합하여 입력해야 합니다.'
+      );
+    } else if (!validEmail) {
+      alert('[ 이메일 ] 은(는) 공백없이 @/. 을 조합하여 입력해야 합니다');
+    } else {
+      fetch('http://10.58.0.27:8000/users/signup', {
+        method: 'POST',
+        body: JSON.stringify({
+          username: id,
+          password: pw,
+          email: email,
+        }),
+      })
+        .then(res => res.json())
+        .then(res => {
+          const { message } = res;
+          if (res.message === 'Success') {
+            navigate('/login');
+          } else {
+            alert(LOGIN_MESSAGE[`${message}`]);
+          }
```

Comment:
```suggestion
          const { message } = res;
          if (message === 'Success') {
            navigate('/login');
          } else {
            alert(LOGIN_MESSAGE[message]);
          }
```

* 구조분해할당 하셨다면, res.message도 mesaage로 쓸 수 있겠네요
* 템플릿리터럴은 적용하지않아도 됩니다. 계산된속성명 혹은 computed property 검색 해 보세요

Reply:
수정했습니다!